data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/BinaryOperator.java

```diff
@@ -18,11 +18,6 @@ protected BinaryOperator(final int symbol, final int shouldEvaluateRuleIndex) {
         displayName = DataPrepperExpressionParser.VOCABULARY.getDisplayName(symbol);
     }
 
-    @Override
-    public int getNumberOfOperands() {
-        return NUMBER_OF_OPERANDS;
```

Comment:
Please remove line 9 in this file now.

We should leverage a single source of truth for a class. The evaluate method uses a hardcoded number to enforce the number of operands. Can we update the `evaluate` method to leverage the default method now as well?